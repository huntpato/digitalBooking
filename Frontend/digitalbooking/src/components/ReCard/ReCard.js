import React, { Fragment, useState, useEffect } from "react";
import styles from "./ReCard.module.css";
import { ReviewCount } from "../ReviewCount/ReviewCount";
import { ReTitle } from "../ReTitle/ReTitle";
import { ReServices } from "../ReServices/ReServices";
import { Link } from "react-router-dom";
import { Maps } from "../Maps/Maps";

export const ReCard = (props) => {
  const { type, title, img, alt, text, rate, id, services, latlong } = props;
  const link = `/product/${id}`;
  const {
    recommendationCards,
    location,
    reCardTitle,
    container,
    description,
    recommendationMap,
    mapHeader,
  } = styles;
  const [mapVisible, setMapVisible] = useState(false);
  const [lat, setLat] = useState("-34.70");
  const [long, setLong] = useState("-58.37"); 

  const handleClick = () => {
    setMapVisible(false);
  };

  const showMap = () => {
    setMapVisible(true);
  };

  return (
    <Fragment>
      {!mapVisible && (
        <article className={recommendationCards}>
          <img src={img} alt={alt} />
          <div className={container}>
            <div className={reCardTitle}>
              <ReTitle type={type} title={title} />
              <ReviewCount rate={rate} />
            </div>
            <div className={location}>
              <p>A 940 mts del centro</p>
              <h6 onClick={ showMap }>MOSTRAR EN EL MAPA</h6>
            </div>
            <ReServices locServices={services} />
            <div className={description}>
              <p>{text}</p>
            </div>
            <Link to={link}>
              {" "}
              <button>Ver m√°s</button>
            </Link>
          </div>
        </article>
      )}
      {mapVisible && (
        <article className={recommendationMap}>
          <div className={mapHeader}>
            <h6>{title}</h6>
            <span>{location}</span>
            <p onClick={handleClick}>X</p>
          </div>
          <Maps
            lat={latlong[0]}
            long={latlong[1]}
            name={title}
            city={location}
            width="92%"
            height="100%"
            margin="4%"
          />
        </article>
      )}
    </Fragment>
  );
};


