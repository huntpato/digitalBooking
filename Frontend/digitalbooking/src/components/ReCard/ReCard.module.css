.recommendationCards {
  background-color: var(--background-white);
  border-radius: 6px;
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
  display: flex;
  margin-top: 10px;
  margin-bottom: 12px;
  flex-direction: row;
  justify-content: space-between;
  height: 246px;
  width: 49%;
  -webkit-animation: fade-in 1.2s cubic-bezier(0.390, 0.575, 0.565, 1.000) both;
  animation: fade-in 1.2s cubic-bezier(0.390, 0.575, 0.565, 1.000) both;
}

.recommendationMap {
  background-color: var(--background-white);
  border-radius: 6px;
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
  display: flex;
  margin-top: 10px;
  margin-bottom: 12px;
  flex-direction: column;
  justify-content: space-between;
  width: 49%;
  -webkit-animation: flip-in-ver-right 0.5s cubic-bezier(0.25, 0.46, 0.45, 0.94)
    both;
  animation: flip-in-ver-right 0.5s cubic-bezier(0.25, 0.46, 0.45, 0.94) both;
}

.mapHeader{
  align-content: center;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  margin: 10px 25px 0px 25px;
  
}

.recommendationMap p{
  cursor: pointer;
}

.container {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  margin-top: 0%;
  margin-bottom: 10px;
  margin-right: 2%;
  width: 55%;
}

.recommendationCards img {
  border-radius: 5px;
  object-fit: cover;
  height: 100%;
  width: 40%;
}

.recommendationCards p {
  font-size: 11px;
  margin-top: 10px;
}

.recommendationCards button {
  background-color: var(--principal-color);
  border-radius: 5px;
  border: none;
  color: var(--background-white);
  cursor: pointer;
  height: 30px;
  width: 100%;
}

.recommendationCards button:hover {
  background-color: var(--principal-color);
  color: var(--background-white);
  opacity: 2;
} 

.recommendationCards button:active {
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.15);
  transform: scale(0.98);
}

.location {
  align-items: baseline;
  display: flex;
  flex-direction: row;
  margin-bottom: 8px;
}

.location h6{
  color: var(--principal-color);
  cursor: pointer;
}

.location p {
  font-size: 12px;
  margin-right: 5px;
}

.location p::before {
  color: var(--color-typo);
  content: "\f3c5";
  font-family: "Font Awesome 5 Free";
  font-size: 14px;
  font-weight: 700;
  margin-right: 2px;
}

.location a {
  color: var(--principal-color);
  font-size: 12px;
  font-weight: 700;
}

.reCardTitle {
  align-items: baseline;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  margin-top: 10px;
  width: 100%;
}

.description p {
  margin-bottom: 10px;
}

.recommendationMap span{
  display: none;
}



@media (max-width: 768px) {
  
  .recommendationCards {
    width: 100%;
    margin-bottom: 17px;
  }

  .recommendationMap {
    width: 100%;
    margin-bottom: 17px;
    position:relative;
    height: 246px;
  }

}

@media (max-width: 480px) {

  .container {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    margin-top: 0%;
    margin-bottom: 15px;
    margin-right: 0%;
    margin-left: 2%;
    width: 96%;
  }

  .recommendationMap{
    height: 490px;
    width: 100%;
    padding-bottom: 25px;
  }

  .recommendationMap h6{
    position: relative;
    font-size: 16px;
  }

  .recommendationMap span{
    font-size: 10px;
  }

  .recommendationCards {
    flex-direction: column;
    justify-content: center;
    height: 490px;
  }

  .recommendationCards img {
    border-radius: 0px;
    border-radius: 5px 5px 0px 0px;
    height: 300px;
    width: 100%;
  }

  .recommendationCards button {
    width: 100%;
  }

  .reCardTitle {
    margin-top: 0px;
  }
}

@-webkit-keyframes fade-in {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}
@keyframes fade-in {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}

@-webkit-keyframes flip-in-ver-right {
  0% {
    -webkit-transform: rotateY(-80deg);
    transform: rotateY(-80deg);
    opacity: 0;
  }
  100% {
    -webkit-transform: rotateY(0);
    transform: rotateY(0);
    opacity: 1;
  }
}
@keyframes flip-in-ver-right {
  0% {
    -webkit-transform: rotateY(-80deg);
    transform: rotateY(-80deg);
    opacity: 0;
  }
  100% {
    -webkit-transform: rotateY(0);
    transform: rotateY(0);
    opacity: 1;
  }
}
