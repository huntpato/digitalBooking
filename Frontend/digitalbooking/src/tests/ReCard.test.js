import React from 'react';
import { shallow } from 'enzyme';
import { ReCard } from '../components/ReCard/ReCard'
import '@testing-library/jest-dom'
import { expect } from '@jest/globals';

describe('Pruebas en componente  ReCard />', ()=> {
    const type = "hotel";
    const title = "Hotel Hermitage";
    const img = "https://images.unsplash.com/photo-1440778303588-435521a205bc?ixid=MnwxMjA3fDB8MHxzZWFyY2h8NXx8aG9saWRheXxlbnwwfHwwfHw%3D&ixlib=rb-1.2.1&auto=format&fit=crop&w=500&q=60";
    const alt =  "Hotel Hermitage"
    const text=  " Lorem ipsum dolor sit amet consectetur adipiscing elit"
    const rate = 9;  
    const id= 1 
    const wrapper= shallow( <ReCard type= {type} title={title} img={img} alt={alt} text={text} rate= {rate} id={id} />);

    test('Comprobar que  ReCard /> renderiza correctamente.', () => {
    
        expect( wrapper ).toMatchSnapshot();
      
    })

    test('Verificar que el componente carga la clase de CSS correctamente', () => {

        const article = wrapper.find('article');
        expect(article.hasClass('recommendationCards')).toBe(true);
    })

    test('Verificar que el componente renderiza la imagen correcta', () => {
        
        expect (wrapper.find("img").at(0).prop("src")).toEqual(img);

    })

    
    test('Verificar que la imagen contiene el texto alternativo enviado por props', () => {
        
        expect (wrapper.find("img").at(0).prop("alt")).toEqual("Hotel Hermitage");

    })

    test('Verificar que se renderiza la card con el titulo enviado', () => {
        
        expect (wrapper.find("ReTitle").prop("title")).toEqual("Hotel Hermitage");

    })

    test('Verificar que se renderiza la card con la calificacion enviada', () => {
        
        expect (wrapper.find("ReviewCount").prop("rate")).toEqual(9);

    })


    
})
