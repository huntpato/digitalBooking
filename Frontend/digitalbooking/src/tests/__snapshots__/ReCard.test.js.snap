// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente  ReCard /> Comprobar que  ReCard /> renderiza correctamente. 1`] = `
<Fragment>
  <article
    className="recommendationCards"
  >
    <img
      alt="Hotel Hermitage"
      src="https://images.unsplash.com/photo-1440778303588-435521a205bc?ixid=MnwxMjA3fDB8MHxzZWFyY2h8NXx8aG9saWRheXxlbnwwfHwwfHw%3D&ixlib=rb-1.2.1&auto=format&fit=crop&w=500&q=60"
    />
    <div
      className="container"
    >
      <div
        className="reCardTitle"
      >
        <ReTitle
          title="Hotel Hermitage"
          type="hotel"
        />
        <ReviewCount
          rate={9}
        />
      </div>
      <div
        className="location"
      >
        <p>
          A 940 mts del centro
        </p>
        <h6
          onClick={[Function]}
        >
          MOSTRAR EN EL MAPA
        </h6>
      </div>
      <ReServices
        locServices={
          Array [
            "Wifi",
            "kitchen",
            " parking",
            "airconditioning",
          ]
        }
      />
      <div
        className="description"
      >
        <p>
           Lorem ipsum dolor sit amet consectetur adipiscing elit
        </p>
      </div>
      <Link
        to="/product/1"
      >
         
        <button>
          Ver m√°s
        </button>
      </Link>
    </div>
  </article>
</Fragment>
`;
